src/application/service/fact.go

```diff
@@ -16,25 +15,25 @@ import (
 
 type FactService interface {
 	GetById(uuid.UUID) (domain.Fact, error)
-	GetBinaryById(pgx.Tx, uuid.UUID) (io.ReadSeekCloser, error)
-	GetLatestByFields(pgx.Tx, [][]string) (domain.Fact, error)
-	GetByFields(pgx.Tx, [][]string) ([]*domain.Fact, error)
+	GetBinaryByIdAndTx(pgx.Tx, uuid.UUID) (io.ReadSeekCloser, error)
+	GetLatestByFields([][]string) (domain.Fact, error)
```

Comment:
`GetLatestByFields()` and `GetByFields()` need to take a transaction so we can avoid a race condition while [checking action runnability](https://github.com/input-output-hk/cicero/blob/cfa7721338f548e8a0b94162c5bdf8e4fed54025/src/application/service/action.go#L234).

Reply:
Alternative: It would be possible to only take a transaction in `FactRepository.GetLatest{ByFields}()` because the `actionService` uses that [directly](https://github.com/input-output-hk/cicero/blob/cfa7721338f548e8a0b94162c5bdf8e4fed54025/src/application/service/action.go#L34) instead of a `FactService`. This way the `FactService` could start a transaction itself and pass that to the `FactRepository`. But I don't really like that since it just hides the transaction instead of removing it.

Reply:
After restoring to previous state, I forgot to push this change.

> Alternative: It would be possible to only take a transaction in FactRepository.GetLatest{ByFields}() because the actionService uses that directly instead of a FactService. This way the ActionService could start a transaction itself and pass that to the FactRepository. But I don't really like that since it just hides the transaction instead of removing it.

I'm totally agree, you hide the transaction